docs/designs/011-kubernetes-object-monitor.md

```diff
@@ -0,0 +1,267 @@
+# ADR-011: Kubernetes Object Monitor
+
+## Problem
+
+Although NVSentinel has a pluggable architecture, the barrier to adding new monitors is high. Users must write, test, and deploy a complete health monitor to integrate existing health checks that don't fall into current monitor categories (e.g., storage checks). Many of these existing checks already manifest issues by modifying Kubernetes objects such as node conditions, taints, labels, and annotations.
+
+## Solution 
+
+Develop a policy-driven Kubernetes health monitor that is configurable via CEL policies and publishes health events based on rule matches. This allows users to integrate existing health checks without writing new health monitors.
+
+
+An example of such configuration would be:
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kubernetes-health-monitor
+  namespace: nvsentinel
+data:
+  policies.yaml: |
+    policies:
+      # Node condition-based policy
+      # Node not ready for more than 2 hours
+      - name: GPUNodeNotReady
+        enabled: true
+        resourceType: Node
```

Comment:
Don't you also need the group and version too?

Reply:
Assumption here was it would map to node type, but it would be better to have full GVK spec